exp/tools/serpapi/internal/serpapiClient.go

```diff
@@ -0,0 +1,109 @@
+package internal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+var ErrMissingToken = errors.New("missing the OpenAI API key, set it in the SERPAPI_API_KEY environment variable")
+var NoGoodResult = "No good search result found"
+
+type SerpapiClient struct {
+	apiKey  string
+	baseURL string
+}
+
+func New() (*SerpapiClient, error) {
+	apiKey := os.Getenv("SERPAPI_API_KEY")
+	if apiKey == "" {
+		return nil, ErrMissingToken
+	}
+
+	return &SerpapiClient{
+			apiKey:  apiKey,
+			baseURL: "https://serpapi.com/search",
+		},
+		nil
+}
+
+func (s *SerpapiClient) Search(query string) (string, error) {
+	params := make(url.Values)
+	query = strings.ReplaceAll(query, " ", "+")
+	params.Add("q", query)
+	params.Add("google_domain", "google.com")
+	params.Add("gl", "us")
+	params.Add("hl", "en")
+	params.Add("api_key", s.apiKey)
+
+	reqURL := fmt.Sprintf("%s?%s", s.baseURL, params.Encode())
+	resp, err := http.Get(reqURL)
+	if err != nil {
+		log.Printf("Error: %s\n", err)
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Printf("Error: %s\n", err)
+		return "", err
+	}
+	var result map[string]interface{}
```

Comment:
Is that possible to define the response body of serpapi into a standalone struct?

Reply:
mmmm its i think it would be very complex, Whould try to send the complete response to chatGTP and ask for it hahahahaha